addons/sale_product_configurator/views/templates.xml

```diff
@@ -93,22 +97,31 @@
                 <input type="hidden" class="product_template_id" t-att-value="product.product_tmpl_id.id"/>
                 <input type="hidden" class="product_id" t-att-value="product_variant.id"/>
                 <td class='td-img'>
-                    <img t-if="product_variant" t-att-src="'/web/image/product.product/%s/image_128' % product_variant.id" alt="Product Image"/>
-                    <img t-else="" t-att-src="'/web/image/product.template/%s/image_128' % product.id" alt="Product Image"/>
+                    <t t-call="website_sale.shop_product_carousel"/>
                 </td>
                 <td class='td-product_name'>
-                    <strong t-esc="combination_info['display_name']"/>
+                    <strong class="product-name product_display_name" t-esc="combination_info['display_name']"/>
                     <div class="text-muted small">
                         <div t-field="product.description_sale"/>
                         <div class="js_attributes"/>
                     </div>
+                    <div>
+                        <t t-if="product.product_tmpl_id and not combination">
+                            <t t-set="combination" t-value="product.product_tmpl_id._get_first_possible_combination()"/>
+                        </t>
+                        <t t-if="combination" t-call="sale.variants">
+                            <t t-set="ul_class" t-value="'flex-column'" />
+                            <t t-set="product" t-value="product.product_tmpl_id"/>
+                        </t>
+                        <t t-else="">
+                            <ul class="d-none js_add_cart_variants" t-att-data-attribute_exclusions="{'exclusions: []'}"/>
+                        </t>
+                    </div>
                 </td>
                 <td class="text-center td-qty">
                     <t t-call='sale_product_configurator.product_quantity_config' />
                 </td>
                 <td class="text-center td-price" name="price">
-                    <ul class="d-none js_add_cart_variants" t-att-data-attribute_exclusions="{'exclusions: []'}"></ul>
-                    <div class="d-none oe_unchanged_value_ids" t-att-data-unchanged_value_ids="variant_values" ></div>
```

Comment:
here you remove `unchanged_value_ids` data attribute and `oe_unchanged_value_ids` class, which leave dead code -> javascript using it [here](https://github.com/odoo-dev/odoo/blame/b79202a9c8cacb7d2b42ceeaafe4fcbf524a0aa2/addons/sale/static/src/js/variant_mixin.js#L289)
To check if wanted or not, is it is, delete the dead code

Reply:
Ah yes, it doesn't make sense to keep that logic in the JS file, there is no longer a need to store the original main product values since it can still get modified and the values extracted from the modal.